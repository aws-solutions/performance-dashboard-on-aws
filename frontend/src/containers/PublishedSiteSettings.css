.margin-top-2-important {
  margin-top: 16px !important;
}

.published-site p {
  margin: 0;
}
