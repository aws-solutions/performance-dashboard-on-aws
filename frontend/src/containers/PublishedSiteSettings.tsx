/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { useHistory } from "react-router-dom";
import { useHomepage, useSettings } from "../hooks";
import SettingsLayout from "../layouts/Settings";
import Spinner from "../components/Spinner";
import Button from "../components/Button";
import MarkdownRender from "../components/MarkdownRender";
import Link from "../components/Link";
import { useTranslation } from "react-i18next";
import "./PublishedSiteSettings.css";

function PublishedSiteSettings() {
    const history = useHistory();
    const { homepage, loading } = useHomepage();
    const { settings, loadingSettings } = useSettings();
    const { t } = useTranslation();

    const onContentEdit = () => {
        history.push("/admin/settings/publishedsite/contentedit");
    };

    const onNavbarEdit = () => {
        history.push("/admin/settings/publishedsite/navbaredit");
    };

    const onAnalyticsEdit = () => {
        history.push("/admin/settings/publishedsite/analyticsedit");
    };

    return (
        <SettingsLayout>
            <h1>{t("PublishedSiteSettings.PublishedSite")}</h1>

            <p>
                {t("PublishedSiteSettings.HeaderDescription")}{" "}
                <Link
                    target="_blank"
                    to="/"
                    external
                    ariaLabel={`${t("PublishedSiteSettings.ViewPublishedSite")} ${t(
                        "ARIA.OpenInNewTab",
                    )}`}
                >
                    {t("PublishedSiteSettings.ViewPublishedSite")}
                </Link>
            </p>

            <h2 className="margin-top-2-important">{t("PublishedSiteSettings.NavigationBar")}</h2>

            <p>{t("PublishedSiteSettings.NavagationBarDescription")}</p>

            <div className="grid-row margin-top-0-important">
                <div className="grid-col flex-9">
                    <h3>{t("PublishedSiteSettings.Title")}</h3>
                </div>
                <div className="grid-col flex-3 text-right">
                    <Button className="margin-top-2" variant="outline" onClick={onNavbarEdit}>
                        {t("Edit")}
                    </Button>
                </div>
            </div>

            {loadingSettings ? (
                <Spinner className="margin-top-3 text-center" label={t("LoadingSpinnerLabel")} />
            ) : (
                <div className="grid-row margin-top-0-important margin-bottom-4">
                    <div className="grid-col flex-9">
                        <div className="published-site font-sans-lg">
                            <MarkdownRender source={settings.navbarTitle} />
                        </div>
                    </div>
                    <div className="grid-col flex-3 text-right"></div>
                </div>
            )}

            <div className="grid-row margin-top-0-important">
                <div className="grid-col flex-9">
                    <h3>{t("PublishedSiteSettings.ContactEmailAddressLabel")}</h3>
                </div>
            </div>

            {loadingSettings ? (
                <Spinner className="margin-top-3 text-center" label={t("LoadingSpinnerLabel")} />
            ) : (
                <div className="grid-row margin-top-0-important margin-bottom-4">
                    <div className="grid-col flex-9">
                        <div className="published-site font-sans-lg">
                            <MarkdownRender
                                source={
                                    settings.contactEmailAddress
                                        ? settings.contactEmailAddress
                                        : "-"
                                }
                            />
                        </div>
                    </div>
                    <div className="grid-col flex-3 text-right"></div>
                </div>
            )}

            <div className="grid-row margin-top-0-important">
                <div className="grid-col flex-9">
                    <h3>{t("PublishedSiteSettings.ContactUsLabel")}</h3>
                </div>
            </div>

            {loadingSettings ? (
                <Spinner className="margin-top-3 text-center" label={t("LoadingSpinnerLabel")} />
            ) : (
                <div className="grid-row margin-top-0-important margin-bottom-4">
                    <div className="grid-col flex-9">
                        <div className="published-site font-sans-lg">
                            <MarkdownRender
                                source={settings.contactUsContent ? settings.contactUsContent : "-"}
                            />
                        </div>
                    </div>
                    <div className="grid-col flex-3 text-right"></div>
                </div>
            )}

            <hr
                style={{
                    border: "none",
                    height: "1px",
                    backgroundColor: "#dfe1e2",
                    margin: "2rem 0",
                }}
            />

            <h2 className="margin-top-2-important">
                {t("PublishedSiteSettings.HomepageContentHeader")}
            </h2>

            <p>{t("PublishedSiteSettings.HomepageContentDescription")}</p>

            <div className="grid-row margin-top-0-important">
                <div className="grid-col flex-9">
                    <h3>{t("PublishedSiteSettings.Headline")}</h3>
                </div>
                <div className="grid-col flex-3 text-right">
                    <Button className="margin-top-2" variant="outline" onClick={onContentEdit}>
                        {t("Edit")}
                    </Button>
                </div>
            </div>

            {loading ? (
                <Spinner className="margin-top-3 text-center" label={t("LoadingSpinnerLabel")} />
            ) : (
                <div className="grid-row margin-top-0-important">
                    <div className="grid-col flex-9">
                        <div
                            className="published-site font-sans-lg"
                            data-testid="published-site-headline"
                        >
                            <MarkdownRender source={homepage.title} />
                        </div>
                        <div className="grid-col flex-9">
                            <h3>{t("Description")}</h3>
                        </div>
                        <div className="font-sans-lg">
                            <MarkdownRender source={homepage.description} />
                        </div>
                    </div>
                    <div className="grid-col flex-3 text-right"></div>
                </div>
            )}

            <hr
                style={{
                    border: "none",
                    height: "1px",
                    backgroundColor: "#dfe1e2",
                    margin: "2rem 0",
                }}
            />

            <h2 className="margin-top-2-important">{t("PublishedSiteSettings.Analytics")}</h2>

            {loading ? (
                <Spinner className="margin-top-3 text-center" label={t("LoadingSpinnerLabel")} />
            ) : (
                <div className="grid-row margin-top-3-important">
                    <div className="grid-col flex-9 text-left">
                        {t("PublishedSiteSettings.AnalyticsDescription")}
                    </div>
                    <div className="grid-col flex-3 text-right">
                        <Button
                            className="margin-top-0"
                            variant="outline"
                            onClick={onAnalyticsEdit}
                        >
                            {t("Edit")}
                        </Button>
                    </div>
                </div>
            )}
        </SettingsLayout>
    );
}

export default PublishedSiteSettings;
