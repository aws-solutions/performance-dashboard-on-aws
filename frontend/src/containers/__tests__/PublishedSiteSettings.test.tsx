/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import PublishedSiteSettings from "../PublishedSiteSettings";

jest.mock("../../hooks");

test("renders the title", async () => {
    const { getByRole } = render(<PublishedSiteSettings />, {
        wrapper: MemoryRouter,
    });
    expect(getByRole("heading", { name: "Published site" })).toBeInTheDocument();
});

test("renders the navbar title", async () => {
    const { getByRole } = render(<PublishedSiteSettings />, {
        wrapper: MemoryRouter,
    });
    expect(getByRole("heading", { name: "Navigation bar" })).toBeInTheDocument();
});

test("renders the navigation bar title", async () => {
    const { getByText } = render(<PublishedSiteSettings />, {
        wrapper: MemoryRouter,
    });
    expect(getByText("Performance Dashboard")).toBeInTheDocument();
});

test("renders the homepage title", async () => {
    const { getByRole } = render(<PublishedSiteSettings />, {
        wrapper: MemoryRouter,
    });
    expect(getByRole("heading", { name: "Homepage content" })).toBeInTheDocument();
});

test("renders the homepage description", async () => {
    const { getByText } = render(<PublishedSiteSettings />, {
        wrapper: MemoryRouter,
    });
    expect(
        getByText(
            "These components appear on the homepage of your published site and explain what your published site is about.",
        ),
    ).toBeInTheDocument();
});

test("renders the headline header", async () => {
    const { getByText } = render(<PublishedSiteSettings />, {
        wrapper: MemoryRouter,
    });
    expect(getByText("Headline")).toBeInTheDocument();
});

test("renders the headline statement", async () => {
    const { getByText } = render(<PublishedSiteSettings />, {
        wrapper: MemoryRouter,
    });
    expect(getByText("Kingdom of Wakanda")).toBeInTheDocument();
});

test("renders the analytics section", async () => {
    const { getByText } = render(<PublishedSiteSettings />, {
        wrapper: MemoryRouter,
    });
    expect(getByText("Web analytics")).toBeInTheDocument();
});

test("renders a button to edit", async () => {
    const { getAllByRole } = render(<PublishedSiteSettings />, {
        wrapper: MemoryRouter,
    });
    const buttons = getAllByRole("button", { name: "Edit" });
    expect(buttons[0]).toBeInTheDocument();
    expect(buttons[1]).toBeInTheDocument();
});

test("published site settings should match snapshot", async () => {
    const wrapper = render(<PublishedSiteSettings />, {
        wrapper: MemoryRouter,
    });
    expect(wrapper.container).toMatchSnapshot();
});
