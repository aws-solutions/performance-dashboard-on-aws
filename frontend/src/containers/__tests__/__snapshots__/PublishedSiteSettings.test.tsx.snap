// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`published site settings should match snapshot 1`] = `
<div>
  <div
    class="usa-section padding-top-3"
  >
    <div
      class="grid-container padding-x-0"
    >
      <div
        class="grid-row grid-gap"
      >
        <div
          class="desktop:grid-col-3"
        >
          <nav
            aria-label="Left navigation"
          >
            <ul
              class="usa-sidenav"
            >
              <li
                class="usa-sidenav__item"
              >
                <a
                  class="usa-current"
                  href="/admin/settings"
                >
                  Settings
                </a>
                <ul
                  class="usa-sidenav__sublist"
                >
                  <li
                    class="usa-sidenav__item"
                  >
                    <a
                      aria-current="page"
                      class="usa-current"
                      href="/admin/settings/topicarea"
                    >
                      Topic areas
                    </a>
                  </li>
                </ul>
                <ul
                  class="usa-sidenav__sublist"
                >
                  <li
                    class="usa-sidenav__item"
                  >
                    <a
                      class=""
                      href="/admin/settings/publishingguidance"
                    >
                      Publishing guidance
                    </a>
                  </li>
                </ul>
                <ul
                  class="usa-sidenav__sublist"
                >
                  <li
                    class="usa-sidenav__item"
                  >
                    <a
                      class=""
                      href="/admin/settings/brandingandstyling"
                    >
                      Branding and styling
                    </a>
                  </li>
                </ul>
                <ul
                  class="usa-sidenav__sublist"
                >
                  <li
                    class="usa-sidenav__item"
                  >
                    <a
                      class=""
                      href="/admin/settings/publishedsite"
                    >
                      Published site
                    </a>
                  </li>
                </ul>
                <ul
                  class="usa-sidenav__sublist"
                >
                  <li
                    class="usa-sidenav__item"
                  >
                    <a
                      class=""
                      href="/admin/settings/dateformat"
                    >
                      Date and time format
                    </a>
                  </li>
                </ul>
                <ul
                  class="usa-sidenav__sublist"
                >
                  <li
                    class="usa-sidenav__item"
                  >
                    <a
                      class=""
                      href="/admin/settings/adminsite"
                    >
                      Admin site
                    </a>
                  </li>
                </ul>
              </li>
            </ul>
          </nav>
        </div>
        <div
          class="desktop:grid-col-9 usa-prose mobile-margin-top-2"
          id="main-content"
        >
          <h1>
            Published site
          </h1>
          <p>
            Customize your published site. Your saved edits will take effect immediately on the published site.
             
            <a
              aria-label="View published site (opens in a new tab)"
              class="usa-link action-link"
              href="/"
              target="_blank"
            >
              <span>
                View published site
              </span>
              <svg
                aria-hidden="false"
                aria-label="(opens in a new tab)"
                class="svg-inline--fa fa-external-link-alt fa-w-16 fa-xs margin-left-05"
                data-icon="external-link-alt"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </p>
          <h2
            class="margin-top-2-important"
          >
            Navigation bar
          </h2>
          <p>
            You can customize the header and navigation of the published site.
          </p>
          <div
            class="grid-row margin-top-0-important"
          >
            <div
              class="grid-col flex-9"
            >
              <h3>
                Title
              </h3>
            </div>
            <div
              class="grid-col flex-3 text-right"
            >
              <button
                class="usa-button usa-button--outline margin-top-2"
              >
                Edit
              </button>
            </div>
          </div>
          <div
            class="grid-row margin-top-0-important margin-bottom-4"
          >
            <div
              class="grid-col flex-9"
            >
              <div
                class="published-site font-sans-lg"
              >
                <div
                  class="Markdown undefined"
                >
                  <p>
                    Performance Dashboard
                  </p>
                </div>
              </div>
            </div>
            <div
              class="grid-col flex-3 text-right"
            />
          </div>
          <div
            class="grid-row margin-top-0-important"
          >
            <div
              class="grid-col flex-9"
            >
              <h3>
                Contact email address
              </h3>
            </div>
          </div>
          <div
            class="grid-row margin-top-0-important margin-bottom-4"
          >
            <div
              class="grid-col flex-9"
            >
              <div
                class="published-site font-sans-lg"
              >
                <div
                  class="Markdown undefined"
                >
                  <p>
                    contact@example.com
                  </p>
                </div>
              </div>
            </div>
            <div
              class="grid-col flex-3 text-right"
            />
          </div>
          <div
            class="grid-row margin-top-0-important"
          >
            <div
              class="grid-col flex-9"
            >
              <h3>
                Contact us
              </h3>
            </div>
          </div>
          <div
            class="grid-row margin-top-0-important margin-bottom-4"
          >
            <div
              class="grid-col flex-9"
            >
              <div
                class="published-site font-sans-lg"
              >
                <div
                  class="Markdown undefined"
                >
                  <ul>
                    <li />
                  </ul>
                </div>
              </div>
            </div>
            <div
              class="grid-col flex-3 text-right"
            />
          </div>
          <hr
            style="height: 1px; background-color: rgb(223, 225, 226); margin: 2rem 0px;"
          />
          <h2
            class="margin-top-2-important"
          >
            Homepage content
          </h2>
          <p>
            These components appear on the homepage of your published site and explain what your published site is about.
          </p>
          <div
            class="grid-row margin-top-0-important"
          >
            <div
              class="grid-col flex-9"
            >
              <h3>
                Headline
              </h3>
            </div>
            <div
              class="grid-col flex-3 text-right"
            >
              <button
                class="usa-button usa-button--outline margin-top-2"
              >
                Edit
              </button>
            </div>
          </div>
          <div
            class="grid-row margin-top-0-important"
          >
            <div
              class="grid-col flex-9"
            >
              <div
                class="published-site font-sans-lg"
                data-testid="published-site-headline"
              >
                <div
                  class="Markdown undefined"
                >
                  <p>
                    Kingdom of Wakanda
                  </p>
                </div>
              </div>
              <div
                class="grid-col flex-9"
              >
                <h3>
                  Description
                </h3>
              </div>
              <div
                class="font-sans-lg"
              >
                <div
                  class="Markdown undefined"
                >
                  <p>
                    Welcome to our dashboard
                  </p>
                </div>
              </div>
            </div>
            <div
              class="grid-col flex-3 text-right"
            />
          </div>
          <hr
            style="height: 1px; background-color: rgb(223, 225, 226); margin: 2rem 0px;"
          />
          <h2
            class="margin-top-2-important"
          >
            Web analytics
          </h2>
          <div
            class="grid-row margin-top-3-important"
          >
            <div
              class="grid-col flex-9 text-left"
            >
              Add your web analytics tracking ID to get insights into how your visitors are interacting with Performance Dashboard.
            </div>
            <div
              class="grid-col flex-3 text-right"
            >
              <button
                class="usa-button usa-button--outline margin-top-0"
              >
                Edit
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
